cosmwasm/packages/enclave-ffi-types/src/types.rs

```diff
@@ -92,6 +92,28 @@ pub enum EnclaveError {
     Unknown,
 }
 
+/// This type represents the possible error conditions that can be encountered in the
+/// enclave while authenticating a new node in the network.
+/// cbindgen:prefix-with-name
+#[repr(C)]
+#[derive(Debug, Display, PartialEq, Eq)]
+pub enum NodeAuthResult {
```

Comment:
Do you think we could add errors for specific isvEnclaveQuoteStatus? Like, return "failed to perform registration. Invalid quote status "KEY_REVOKED" " or something

Reply:
hmm... can you point me to the place in the code where this error originates? I'd probably have to fiddle with 1-3 functions to bubble this specific case out to the context where i instantiate variants of this type.
Note that the way i use this type is really high level, and unrelated to how most of the related code just uses `sgx_status_t`

Reply:
https://github.com/enigmampc/SecretNetwork/blob/05787bae238375af0e95662e3006ba56451f9590/cosmwasm/packages/wasmi-runtime/src/registration/cert.rs#L360

If it's not something you can do in 15 minutes or less we can leave it for some other time

Reply:
Yeah, this will be somewhat involved.
I'm not sure I understand what exactly all the structs and enums mean, and the specific error you're referring to happens a couple layers into functions that obfuscate the specific errors that happen. If I change that, I'd need to add a couple of error types, and modify a bunch of other functions that use the same functions that I needed to change in the first place.